docs/core/diagnostics/logging-tracing.md

```diff
@@ -0,0 +1,97 @@
+ms.date: 08/05/2019
+---
+# .NET Core logging and tracing
+
+Logging and tracing are really two names for the same technique. The simple technique has been used since the early days of computers. It simply involves instrumenting an application to write output to be consumed later.
+
+## Reasons to use logging and tracing
+
+This simple technique is surprisingly powerful. It can be used in situations where a debugger fails:
+
+- Issues occurring over long periods of time, can be difficult to debug with a traditional debugger. Logs allow for detailed post-mortem review spanning long periods of time. In contrast, debuggers are constrained to real-time analysis.
+- Multi-threaded applications and distributed applications are often difficult to debug.  Attaching a debugger tends to modify behaviors. Detailed logs can be analyzed as needed to understand complex systems.
+- Issues in distributed applications may arise from a complex interaction between many components and it may not be reasonable to connect a debugger to every part of the system.
+- Many services shouldn't be stalled. Attaching a debugger often causes timeout failures.
+- Issues aren't always foreseen. Logging and tracing are designed for low overhead so that programs can always be recording in case an issue occurs.
+
+## .NET Core APIs
+
+The <xref:System.Console?displayProperty=nameWithType>, <xref:System.Diagnostics.Trace?displayProperty=nameWithType>, and <xref:System.Diagnostics.Debug?displayProperty=nameWithType> classes each provide similar print style APIs convenient for logging.
+
+The choice of which print style API to use is up to you. The key differences are:
+- <xref:System.Console?displayProperty=nameWithType>
+  - Always enabled and always writes to the console.
+  - Useful for information that your customer may need to see in the release.
+  - Because it's the simplest approach, it's often used for ad-hoc temporary debugging. This debug code is often never checked in to source control.
+- <xref:System.Diagnostics.Trace?displayProperty=nameWithType>
+  - Only enabled when `TRACE` is defined.
+  - Writes to attached <xref:System.Diagnostics.Trace.Listeners>, by default the <xref:System.Diagnostics.DefaultTraceListener>.
+  - Use this API when creating logs that will be enabled in most builds.
+- <xref:System.Diagnostics.Debug?displayProperty=nameWithType>
+  - Only enabled when `DEBUG` is defined.
+  - Writes to an attached debugger.
+  - On `*nix` writes to stderr if `COMPlus_DebugWriteToStdErr` is set.
+  - Use this API when creating logs that will be enabled only in debug builds.
+
+- <xref:System.Diagnostics.Tracing.EventSource?displayProperty=nameWithType>
+  - EventSource is the primary root .NET Core tracing API.
+  - Available in all .NET Standard versions.
+  - Only allows tracing serializable objects.
+  - Writes to attached <xref:System.Diagnostics.Tracing.EventListener>s.
+  - .NET Core provides listeners for:
+    - .NET Core's event pipe on all platforms
```

Comment:
I'd prefer to reword `event pipe`  to `EventPipe` for the sake of consistency across documentations.